catris/games/game_base_class.py

```diff
@@ -48,20 +50,24 @@ def __init__(self) -> None:
         self.flashing_lock = asyncio.Lock()
         self.flashing_squares: dict[tuple[int, int], int] = {}
 
-    def pause(self) -> None:
-        self._pause_event.set()
-        self._unpause_event.clear()
-        self.need_render_event.set()
-
-    def unpause(self) -> None:
-        self._pause_event.clear()
-        self._unpause_event.set()
-        self.need_render_event.set()
-
     @property
     def is_paused(self) -> bool:
         return self._pause_event.is_set()
 
+    def toggle_pause(self) -> None:
+        if self.is_paused:
+            self._pause_event.clear()
+            self._unpause_event.set()
+            self._time_spent_in_pause += time.monotonic_ns() - self._last_pause_start
+        else:
+            self._pause_event.set()
+            self._unpause_event.clear()
+            self._last_pause_start = time.monotonic_ns()
+        self.need_render_event.set()
+
+    def get_duration_ns(self) -> int:
+        return time.monotonic_ns() - self._start_time - self._time_spent_in_pause
```

Comment:
This will produce wrong results if called on a paused game.
```suggestion
        assert not self.is_paused
        return time.monotonic_ns() - self._start_time - self._time_spent_in_pause
```